src/Microsoft.IdentityModel.Tokens/SecurityKey.cs

```diff
@@ -37,11 +37,23 @@ namespace Microsoft.IdentityModel.Tokens
     public abstract class SecurityKey
     {
         private CryptoProviderFactory _cryptoProviderFactory;
+        private readonly Lazy<string> _internalId;
 
         internal SecurityKey(SecurityKey key)
         {
             _cryptoProviderFactory = key._cryptoProviderFactory;
             KeyId = key.KeyId;
+            _internalId = new Lazy<string>(() =>
+            {
+                try
+                {
+                    return Base64UrlEncoder.Encode(ComputeJwkThumbprint());
+                }
+                catch
+                {
+                    return string.Empty;
+                }
+            });
```

Comment:
call base? or refactor into a method?

Reply:
I'll create a new private method as cryptoProviderFactory differs in these two constructors.

Reply:
Addressed with https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/1346/commits/0f00581f0a75b4972b983006eda9aad40e728414.